### ISSUE TITLE
'brew ls --cask --versions' behavior does not match 'brew ls'

### ISSUE THREAD

**Issue by JohnStarich** [Homebrew/brew#13139]
### `brew config` output

```shell
➜ ~ brew config
HOMEBREW_VERSION: 3.4.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 56415326bb8c9eccc6e53f3997b4e062b36402af
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 40b2e40ff30421b04c74359a57cc79a7b6c6e226
Core tap last commit: 43 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: nvim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.35.2 => /opt/homebrew/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2-arm64
CLT: 13.3.1.0.1.1648687083
Xcode: N/A
Rosetta 2: false
➜ ~
```

### `brew doctor` output

```shell
➜ ~ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/johnstarich/Library/Python/3.8/bin/powerline-config
➜ ~
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I use a script to detect if certain formulae (and casks) are installed. I've just tried it on a new machine to set it up, and the current behavior confused cask detection a bit.
EDIT: After further investigation, the exit code was correct but the output was wrong.

The full script I use: https://github.com/JohnStarich/dotfiles/blob/d306530c72581954e8f480c4179a555e9d9f0752/brew/install.sh#L71-L110

### What happened (include all command output)?

Here's a minimal example:

```bash
➜ ~ brew ls --versions git              # List an installed formula, succeeds.
git 2.35.2
➜ ~ brew ls --versions sl               # List an uninstalled formula, fails.
➜ ~ brew ls --versions blah             # List a non-existent formula, fails.

➜ ~ brew ls --cask --versions qlstephen # List an installed cask, succeeds.
qlstephen 1.5.1
➜ ~ brew ls --cask --versions slack     # List an uninstalled cask, fails with bad output?
corelocationcli 3.2.0
qlstephen 1.5.1
➜ ~ brew ls --cask --versions blah      # List a non-existent formula, fails with bad output?
corelocationcli 3.2.0
qlstephen 1.5.1
➜ ~
```
Specifically, the last 2 commands produced unexpected results when compared with the usual list.

### What did you expect to happen?

I expected `brew ls --cask` to behave like `brew ls` with formulae that aren't installed.

In other words `brew ls --cask --versions blah` should exit with a non-zero exit code without any output.

It has the expected behavior without the `--versions` flag. I suspect this part of the help text is not implemented for casks:
```
➜ ~ brew ls --help
      ...
      --versions                   ..., or only the specified formulae if
                                   formula are provided.
      ...
➜ ~
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew ls --versions some-non-existent-formula
vs
brew ls --cask --versions some-non-existent-formula
```

**Comment by MikeMcQuaid** [comment]
> Specifically, the last 2 commands produced unexpected results when compared with the usual list.

Agreed.
